### ISSUE TITLE
tf.igamma (lower regularized incomplete Gamma function) returns the incorrect derivative

### ISSUE THREAD

**Issue by erikbern** [tensorflow/tensorflow#17995]
### System information
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**: yes
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Mac OS X 10.11.6
- **TensorFlow installed from (source or binary)**: binary (pypi)
- **TensorFlow version (use command below)**: v1.6.0-0-gd2e24b6039
- **Python version**: 3.6
- **Bazel version**: using a precompiled version, not sure
- **CUDA/cuDNN version**: not using a GPU
- **GPU model and memory**: not using a GPU
- **Exact command to reproduce**: `python igamma_test.py`

### Describe the problem

`tf.igamma`, which is the lower regularized incomplete Gamma function, returns an incorrect derivative with respect to `a`

This is probably very low down on the list of things to fix, but I wanted to highlight it since I spent something like 5 hours trying to understand why my model wasn't converging. I was fitting a Gamma distribution, and deep into the code it turns out that `tf.igamma` doesn't return the right derivative.

I suspect the derivative wrt `a` isn't supported, but I would have much rather seen an exception being thrown. 

My workaround ended up being not fitting `a` with gradient descent, but instead just perturbing it by epsilon (luckily I only had one single value that I tried to fit)

Filing this issue mostly in the hope that anyone in the future doesn't waste the same amount of time that I spent.

### Source code / logs

```
a = tf.placeholder(dtype=tf.float32, shape=[None])
x = tf.placeholder(dtype=tf.float32, shape=[None])
y = tf.igamma(a, x)
y_grad_a = tf.gradients(y, a)  # returns None, should return a tensor
y_grad_x = tf.gradients(y, x)  # returns a tensor
```

**Comment by tensorflowbutler** [comment]
Thank you for your post. We noticed you have not filled out the following field in the issue template. Could you update them if they are relevant in your case, or leave them as N/A? Thanks.
Bazel version
CUDA/cuDNN version
GPU model and memory
Exact command to reproduce

**Comment by tensorflowbutler** [comment]
Nagging Assignee @tatatodd: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @tatatodd: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @tatatodd: It has been 49 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by rlouf** [comment]
I was facing the same issue. The derivative wrt a is indeed not implemented: https://github.com/tensorflow/tensorflow/blob/d8f9538ab48e3c677aaf532769d29bc29a05b76e/tensorflow/python/ops/math_grad.py

**Comment by tensorflowbutler** [comment]
Nagging Assignee @tatatodd: It has been 15 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @tatatodd: It has been 29 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @tatatodd: It has been 45 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @tatatodd: It has been 60 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @tatatodd: It has been 77 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @tatatodd: It has been 93 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @tatatodd: It has been 109 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @tatatodd: It has been 123 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @tatatodd: It has been 138 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @tatatodd: It has been 153 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @tatatodd: It has been 169 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @tatatodd: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by rmothukuru** [comment]
@erikbern,
Sorry for the delayed response. Your code works fine with **`Tensorflow Version 1.15.2`**. Please find [the Gist](https://colab.research.google.com/gist/rmothukuru/b2d7b9b7cf5831876a5bc1a6b9482868/gh_17995.ipynb) of the working code. Thanks!
